st2actions/st2actions/workflows/workflows.py

```diff
@@ -85,6 +93,53 @@ def process(self, message):
             # the database and fail the workflow execution gracefully. In this case,
             # the garbage collector will find and cancel these workflow executions.
             self.fail_workflow_execution(message, e)
+        finally:
+            if type(message) == wf_db_models.WorkflowExecutionDB:
+                self._running_workflows.remove(str(message["action_execution"]))
+            else:
+                try:
+                    self._running_workflows.remove(str(message["parent"]))
+                except ValueError:
+                    pass
+
+    def shutdown(self):
+        super(WorkflowExecutionHandler, self).shutdown()
+
+        if cfg.CONF.workflow_engine.graceful_shutdown:
+
+            coordinator = coordination.get_coordinator()
+            member_ids = []
+            service = "workflow_engine"
```

Comment:
Get the service name from the constant at https://github.com/StackStorm/st2/blob/master/st2actions/st2actions/cmd/workflow_engine.py#L40